agent/api/container/container_test.go

```diff
@@ -298,8 +267,109 @@ func TestInjectV3MetadataEndpoint(t *testing.T) {
 }
 
 func TestShouldCreateWithSSMSecret(t *testing.T) {
-	for _, test := range shouldCreateWithSecretsTests {
+	cases := []struct {
+		in  Container
+		out bool
+	}{
+		{Container{
+			Name:  "myName",
+			Image: "image:tag",
+			Secrets: []Secret{
+				Secret{
+					Provider:  "ssm",
+					Name:      "secret",
+					ValueFrom: "/test/secretName",
+				}},
+		}, true},
+		{Container{
+			Name:    "myName",
+			Image:   "image:tag",
+			Secrets: nil,
+		}, false},
+		{Container{
+			Name:  "myName",
+			Image: "image:tag",
+			Secrets: []Secret{
+				Secret{
+					Provider:  "asm",
+					Name:      "secret",
+					ValueFrom: "/test/secretName",
+				}},
+		}, false},
+	}
+
+	for _, test := range cases {
 		container := test.in
 		assert.Equal(t, test.out, container.ShouldCreateWithSSMSecret())
 	}
-}
\ No newline at end of file
+}
+
+func TestMergeEnvironmentVariables(t *testing.T) {
+	cases := []struct {
+		Name                   string
+		InContainerEnvironment map[string]string
+		InEnvVarMap            map[string]string
+		OutEnvVarMap           map[string]string
+	}{
+		{
+			Name: "merge single item",
+			InContainerEnvironment: map[string]string{
+				"CONFIG1": "config1"},
+			InEnvVarMap: map[string]string{
+				"SECRET1": "secret1"},
+			OutEnvVarMap: map[string]string{
+				"CONFIG1": "config1",
+				"SECRET1": "secret1",
+			},
+		},
+
+		{
+			Name: "merge single item to nil container env var map",
+			InContainerEnvironment: nil,
+			InEnvVarMap: map[string]string{
+				"SECRET1": "secret1"},
+			OutEnvVarMap: map[string]string{
+				"SECRET1": "secret1",
+			},
+		},
+
+		{
+			Name: "merge zero items to existing container env var map",
+			InContainerEnvironment: map[string]string{
+				"CONFIG1": "config1"},
+			InEnvVarMap: map[string]string{},
```

Comment:
Just wonder in what situation the InEnvVarMap will be an empty map? Seems we check whether the container has secrets using **ShouldCreateWithSSMSecret** before calling **MergeEnvironmentVariables**?